### PR TITLE
Add RequestNotFound error

### PR DESCRIPTION
**Pull request by JuannFerrari**
Hi! :wave: I leave a small contribution here, thank you for developing this gem. This error happened to me in a system I'm working on and I use a workaround to handle it, but I thought it might be useful if I fix the error directly in the gem, hope you like it!

refers to https://github.com/fastruby/harvesting/issues/55

------------

### What changes does this PR include?

**Description**

When a user tries to create a new assignment in a project the gem does the next:

`POST /v2/projects/{PROJECT_ID}/user_assignments`

 The problem is that the `PROJECT_ID` could be `nil` (in the worst-case scenario), so the response of Harvest API is an HTML body (in a string) with `404` HTTP code.

Screen Shot 2020-08-21 at 17 36 14

Harvesting tries to parse this HTML as a JSON and fails

```
    def create(entity)
      url = "#{DEFAULT_HOST}/#{entity.path}"
      uri = URI(url)
      response = http_response(:post, uri, body: entity.to_hash)
      entity.attributes = JSON.parse(response.body) #### this line
      entity
    end
```

This raises a `JSONParse::Error` because is trying to parse an HTML response, and this error is not being rescued anywhere and handling it in a proper way. This PR fixes that situation :)

**Summary**

- Added `Harvesting::RequestNotFound` to `errors.rb`.
- Added `Harvesting::RequestNotFound` rescue to `client.rb`.

### CODE REVIEW

File: lib/harvesting/client.rb [hunk 1]
```diff
@@ -148,6 +148,9 @@ def http_response(method, uri, opts = {})
 
       raise Harvesting::AuthenticationError.new(response.to_s) if auth_error?(response)
       raise Harvesting::UnprocessableRequest.new(response.to_s) if response.code.to_i == 422
+      if response.code.to_i == 404
+        raise Harvesting::RequestNotFound.new('The page you were looking for may have been moved or the address misspelled.')
+      end
```
Comment (etagwerker) [inline lib/harvesting/client.rb:153]: @JuannFerrari I think maybe we can add a bit more value to the user if we tell them what URI they were trying to send a request to? 

```suggestion
      raise Harvesting::RequestNotFound.new(uri) if response.code.to_i == 404
```

File: lib/harvesting/errors.rb [hunk 2]
```diff
@@ -4,4 +4,7 @@ class AuthenticationError < StandardError
 
   class UnprocessableRequest < StandardError
   end
+
+  class RequestNotFound < StandardError
+  end
```
Comment (etagwerker) [inline lib/harvesting/errors.rb:9]: @JuannFerrari And then here we could include the string we want to use: 

```suggestion
    def initialize(uri)
      super( "The page you were looking for may have been moved or the address misspelled: #{uri}")
    end
  end
```

**Review by etagwerker — COMMENTED** [review]
@JuannFerrari Looks good, thank you! 👍

I added a suggestion to make it more informative for the end user.

**Comment by JuannFerrari** [comment]
Hey @etagwerker 

I totally agree with your suggestions, I have already made the changes.

Thanks!

**Review by etagwerker — APPROVED** [review]
@JuannFerrari Great, thanks! 💯

**Pull request merged**